applications/StructuralMechanicsApplication/custom_strategies/custom_schemes/explicit_central_differences_scheme.hpp

```diff
@@ -0,0 +1,922 @@
+                current_acceleration[j]  = 0.0;
+                middle_velocity[j]       = 0.0; 
+                
+              }
+              
+              current_velocity[j]      = middle_velocity[j] + (mTime.Previous - mTime.PreviousMiddle) * current_acceleration[j]; //+ actual_velocity;
+              middle_velocity[j]       = current_velocity[j] + (mTime.Middle - mTime.Previous) * current_acceleration[j] ; 
+              current_displacement[j]  = current_displacement[j] + mTime.Delta * middle_velocity[j];      
+              
+              
+          }//for DoF
+
+
+          ////// ROTATION DEGRESS OF FREEDOM
+          if (i->HasDofFor(ROTATION_X))
+          {
+            array_1d<double,3> nodal_inertia     = i->FastGetSolutionStepValue(NODAL_INERTIA);  
+            array_1d<double,3>& current_residual_moment          = i->FastGetSolutionStepValue(MOMENT_RESIDUAL);
+            array_1d<double,3>& current_angular_velocity         = i->FastGetSolutionStepValue(ANGULAR_VELOCITY);
+            array_1d<double,3>& current_rotation                 = i->FastGetSolutionStepValue(ROTATION);
+            array_1d<double,3>& middle_angular_velocity          = i->FastGetSolutionStepValue(MIDDLE_ANGULAR_VELOCITY);
+            array_1d<double,3>& current_angular_acceleration     = i->FastGetSolutionStepValue(ANGULAR_ACCELERATION);
+
+            for (int kk = 0; kk<3; ++kk)
+            {         
+              if (nodal_inertia[kk] > numerical_limit)  current_angular_acceleration[kk] = current_residual_moment[kk] / nodal_inertia[kk];
+              else current_angular_acceleration[kk] = 0.00;
+            }
+            
+
+            DoF = 2;
+            bool Fix_rot[3] = {false, false, false};
+            Fix_rot[0] = (i->pGetDof(ROTATION_X))->IsFixed();
+            Fix_rot[1] = (i->pGetDof(ROTATION_Y))->IsFixed();   
+            
+            
+            if (i->HasDofFor(ROTATION_Z))
+            {
+              DoF = 3;
+              Fix_rot[1] = (i->pGetDof(ROTATION_Z))->IsFixed();  
+            }
+
+            for (int j = 0; j < DoF; j++)
+            {
+              if (Fix_rot[j])
+              {
+                current_angular_acceleration[j] = 0.00;
+                middle_angular_velocity[j] = 0.00;
+              }
+              current_angular_velocity[j]  = middle_angular_velocity[j] + (mTime.Previous - mTime.PreviousMiddle) * current_angular_acceleration[j]; 
+              middle_angular_velocity[j]   = current_angular_velocity[j] + (mTime.Middle - mTime.Previous) * current_angular_acceleration[j] ; 
+              current_rotation[j]          = current_rotation[j] + mTime.Delta * middle_angular_velocity[j];
+            }//for DoF
+          }// Rot DoF
+
+        }//for Node 
+      }//parallel
+
+      mTime.Previous = mTime.Current;
+      mTime.PreviousMiddle = mTime.Middle;
+
+      KRATOS_CATCH("")
+    }
+
+    virtual void SchemeCustomInitialization(ModelPart& rModelPart)
```

Comment:
This function seems to do very similar things to the `Update` function
Can you combine the common stuff?

Reply:
If you refactor the `Update` function then you can use the refactored functions also here from what I can see